
When the PGO tests are run under `buildctl`, SIGINT is ignored and that causes
a test failure. Make the test a bit cleverer and have it check if SIGINT
is going to be seen.

--- a/Lib/test/test_generators.py~	Tue Nov  4 12:46:08 2025
+++ a/Lib/test/test_generators.py	Tue Nov  4 12:46:10 2025
@@ -43,6 +43,8 @@
         except BaseException as _exc:
             exc = _exc
         self.assertIs(type(exc), StopIteration)
+        import signal
+        if signal.getsignal(signal.SIGINT) != signal.SIG_IGN:
-        self.assertEqual(exc.value, "PASSED")
+            self.assertEqual(exc.value, "PASSED") #WS
 
 
